ethereum/core/src/main/java/tech/pegasys/teku/core/ForkChoiceUtil.java

```diff
@@ -385,7 +383,20 @@ public static AttestationProcessingResult on_attestation(
     Attestation attestation = validateableAttestation.getAttestation();
 
     return validateOnAttestation(store, attestation, forkChoiceStrategy)
-        .ifSuccessful(() -> indexAndValidateAttestation(validateableAttestation, maybeTargetState))
+        .ifSuccessful(
+            () -> {
+              if (maybeTargetState.isEmpty()) {
+                return AttestationProcessingResult.UNKNOWN_BLOCK;
+              } else {
+                try {
+                  return is_valid_indexed_attestation(
+                      maybeTargetState.get(), validateableAttestation);
+                } catch (IllegalArgumentException e) {
```

Comment:
(nit) Might be a little cleaner to move this try / catch into `is_valid_indexed_attestation`

Reply:
Done.